extension/src/dashboard/pages/Authentication/LoginForm.tsx

```diff
@@ -0,0 +1,169 @@
+import { parseUser } from 'src/types/User';
+import useUser from 'src/hooks/use-user';
+
+// TODO: make it common ?
+import { useFormErrors, GlobalErrorHandler, FieldErrorsHandler } from 'src/popup/components/Form';
+
+// TODO: make it common ?
+const getGlobalError: GlobalErrorHandler = (error: AxiosError) => {
+  if (!error || !error.response)
+    return;
+
+  const { response: { status, data: { message } } } = error;
+
+  if (status === 401 && message === 'INVALID_CREDENTIALS')
+    return 'Combinaison email / mot de passe non valide';
+
+  if (status === 401 && message === 'EMAIL_NOT_VALIDATED')
+    return 'Votre adresse email n\'a pas été validée, verifiez dans vos spams !';
+};
+
+// TODO: make it common ?
+const getFieldErrors: FieldErrorsHandler = (error: AxiosError) => {
+  if (!error || !error.response || error.response.status !== 400)
+    return;
+
+  const fields = error.response.data;
+
+  const getErrorMessage = (field: string, obj: { [key: string]: string }) => {
+    const constraint = Object.keys(obj)[0];
+
+    if (field === 'email' && constraint === 'isEmail')
+      return 'Format d\'adresse email non valide';
+  };
+
+  return Object.keys(fields)
+    .reduce((errors, field) => ({
+      [field]: getErrorMessage(field, fields[field]),
+      ...errors,
+    }), {});
+};
+
+const useStyles = makeStyles((theme: Theme) => ({
+  container: {
+    '& > div': {
+      marginTop: theme.spacing(2),
+    },
+  },
+  globalError: {
+    textAlign: 'center',
+  },
+  buttonWrapper: {
+    margin: theme.spacing(1),
+    position: 'relative',
+    alignSelf: 'center',
+  },
+  button: {
+    fontWeight: 'bold',
+  },
+  buttonProgress: {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    marginTop: -12,
+    marginLeft: -12,
+  },
+}));
+
+const LoginForm: React.FC = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [valid, setValid] = useState(false);
+
+  const [currentUser, setUser] = useUser();
+  const history = useHistory();
+  const classes = useStyles({});
+
+  const opts: AxiosRequestConfig = { method: 'POST', url: '/api/auth/login' };
+  const [{ data: user, loading, error, status }, login] = useAxios(opts, parseUser, { manual: true });
+
+  const [globalError, errors = {}, resetErrors] = useFormErrors(error, getGlobalError, getFieldErrors);
+
+  useEffect(() => {
+    if (status(200)) {
+      setUser(user);
+      history.push('/');
+    }
+  }, [status, user, setUser, history]);
+
+  useEffect(() => {
+    setValid(![email, password].some((value) => value.length <= 0));
+  }, [email, password]);
+
+  const onSubmit = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+
+    login({
+      data: {
+        email,
+        password,
+      },
+    });
+  };
+
+  if (currentUser)
+    return <Redirect to="/" />;
+
+  return (
+    <form onSubmit={onSubmit}>
+      <FormGroup className={classes.container}>
```

Comment:
Why do we need a `FormGroup` here? (As I understand from the docs, it is used for switches and checkboxes only)

Reply:
You're probably right. I think it's a bad practice I took, using it to wrap `TextField`. But wrap them inside a `Flex` could be more accurate here!